redhat-appstudio/service-provider-integration-operator
# Add support for multiple targets per remote secret.

### What does this PR do?
This modifies the `RemoteSecret` to accept multiple targets.

Pros:
* the model is simple, doesn't require any new logic in persistence, etc.

Cons:
* as with the previous design, this does not allow for overriding values per target (but this is not a requirement at the moment).
* disallows multiple targets per namespace (this is not a big problem, because frankly, that is most probably just a corner case that is not likely to happen in practice).


This is just a draft PR because of the insufficient test coverage and lack of any manual testing.

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/SVPI-477

### How to test this PR?
1. create a remotesecret with 2 targets, each one to a different namespace
2. check that the secret is created in each of the namespaces

```
quay.io/redhat-appstudio/pull-request-builds:spi-controller-593
quay.io/redhat-appstudio/pull-request-builds:spi-oauth-593

or

quay.io/lkrejci/service-provider-integration-operator:svpi-477
quay.io/lkrejci/service-provider-integration-oauth:svpi-477

```
